### DISCUSSION TITLE
Suggestions regarding units in UI

### DISCUSSION THREAD

**Discussion by fhteagle** [electricitymaps/electricitymaps-contrib#5641] [Ideas]
There are times when solar is just getting started for the day in my BA, where the web UI display shows XXX kW / YYY MW. The numbers XXX and YYY are similar enough that if I glanced at it too quickly, I might think it was XXX MW / YYY MW. Consider requiring the units on the same XXX / YYY line to be the same, so that it would instead display 0.X MW / YYY MW (the leading zero is important, please include that). If the value is too small to be meaningful at the MW scale, maybe just display as <0.01 MW / YYY MW.

In the same vein, but a little further, if the maximum value of any item for the geographic area (including whole capacity of the BA) is < 10GW, consider keeping the units in entirely MW. Therefore, 5 GW would be 5000MW. At 10GW and above, everything should be in GW units only. 

I realize that this will hide some data granularity for very small values, but I think that is a worthwhile trade for some consistency in units, and easier to visually scan data.

Thanks

**Comment by VIKTORVAV99** [comment]
Hi and thanks for the suggestion!

While I agree that it could be useful at times I think there is some things we need to consider here first:

- In order for it to always be consistent and ensure it don't change from one minute to the next it's simpler to keep it as is now.

- The current system respect SI prefix formatting to ensure no number grows too large (above 1000) which might require bigger UI changes.

- Changing it would more than likely cause more confusion when there is low production but high installed capacity (like early mornings for solar). Where it could in theory say 1W of 1 000 000 000 W (I realize this would be avoided by truncating the value.)

- As a general rule I don't think we want to change the unit prefix on mostly static values.

I'm happy to discuss it some more but right now I'm leaning towards keeping it as is.

**Comment by fhteagle** [comment]
If there is that many knock on effects and other things that need changed, then maybe my suggestion was not worthwhile after all. Thanks for considering it though.

**Comment by madsnedergaard** [comment]
I think it was a great idea, and thanks for raising the problem of missing the difference :) 
We're currently reworking some of the units while working on #4428, so this might be addressed within the next coming weeks.
